File: Scripts/test_saved_content.mjs
import fs from "node:fs";
import vm from "node:vm";
import assert from "node:assert/strict";
const c = vm.createContext({});
vm.runInContext(fs.readFileSync("docs/saved-content.js", "utf8"), c);
for (const f of JSON.parse(fs.readFileSync("Scripts/fixtures/saved_content.json", "utf8"))) {
  assert.equal(c.SavedContent.title(f.raw, f.body, f.platform), f.want, f.raw);
}
assert.equal(c.SavedContent.breadcrumb("Health", "Green light"), "Health › Green light");
assert.equal(c.SavedContent.excerpt("Sign in to continue"), null);
assert.equal(c.SavedContent.excerpt("  a useful\ncaption  "), "a useful caption");
console.log("Saved content: shared fixtures and breadcrumbs passed.");

File: Scripts/test_saved_content.swift
import Foundation
@main enum SavedContentTests {
    struct Fixture: Decodable { let raw: String; let body: String?; let platform: String; let want: String }
    static func main() throws {
        let data = try Data(contentsOf: URL(fileURLWithPath: "Scripts/fixtures/saved_content.json"))
        let fixtures = try JSONDecoder().decode([Fixture].self, from: data)
        for f in fixtures {
            precondition(SavedContent.title(f.raw, body: f.body, platform: f.platform) == f.want, f.raw)
        }
        precondition(SavedContent.breadcrumb(topic: "Health", subtopic: "Green light") == "Health › Green light")
        precondition(SavedContent.breadcrumb(topic: nil, subtopic: nil).isEmpty)
        precondition(SavedContent.excerpt("  a useful\ncaption  ") == "a useful caption")
        precondition(SavedContent.excerpt("Sign in to continue") == nil)
        print("Saved content: shared fixtures and breadcrumbs passed.")
    }
}
